kpm/spec/kpm/unit/account_spec.rb

```diff
@@ -0,0 +1,347 @@
+      "999999|#{dummy_account_id}|#{dummy_account_id}|willharnet@example.com|Will Harnet||USD|0||||UTC||||||||||||false|2017-04-03T15:50:14.000+0000|demo|2017-04-05T15:01:39.000+0000|Killbill::Stripe::PaymentPlugin|2\n"\
+      "-- account_history record_id|id|target_record_id|external_key|email|name|first_name_length|currency|billing_cycle_day_local|parent_account_id|payment_method_id|is_payment_delegated_to_parent|time_zone|locale|address1|address2|company_name|city|state_or_province|country|postal_code|phone|notes|migrated|is_notified_for_invoices|change_type|created_by|created_date|updated_by|updated_date|tenant_record_id\n"\
+      "999998|#{SecureRandom.uuid}|999999|#{dummy_account_id}|willharnet@example.com|Will Harnet||USD|0||||UTC||||||||||||false|INSERT|demo|2017-04-03T15:50:14.000+0000|demo|2017-04-03T15:50:14.000+0000|2\n"
+    }
+    let(:cols_names) {dummy_data.split("\n")[0].split(" ")[2]}
+    let(:cols_data) {dummy_data.split("\n")[1]}
+    let(:table_name) {dummy_data.split("\n")[0].split(" ")[1]}
+    let(:obfuscating_marker) {:email}
+
+    let(:mysql_cli) {"mysql #{db_name} --user=#{db_username} --password=#{db_password} -e"}
+    let(:mysql_delete_dummy_account) {"DELETE FROM accounts WHERE id = '#{dummy_account_id}'; SELECT ROW_COUNT();"}
+    let(:mysql_delete_dummy_account_history) {"DELETE FROM account_history WHERE external_key = '#{dummy_account_id}'; SELECT ROW_COUNT();"}
+
+  end
+
+  describe '#initialize' do
+    include_context 'account'
+
+    context 'when creating an instance of account class' do
+
+      it 'when initialized with defaults' do
+        expect(described_class.new).to be_an_instance_of(KPM::Account)
+      end
+
+      it 'when initialized with options' do
+        account_class.should be_an_instance_of(KPM::Account)
+        expect(account_class.instance_variable_get(:@killbill_api_key)).to eq(killbill_api_key)
+        expect(account_class.instance_variable_get(:@killbill_api_secrets)).to eq(killbill_api_secrets)
+        expect(account_class.instance_variable_get(:@killbill_user)).to eq(killbill_user)
+        expect(account_class.instance_variable_get(:@killbill_password)).to eq(killbill_password)
+        expect(account_class.instance_variable_get(:@killbill_url)).to eq(url)
+      end
+
+    end
+
+  end
+
+  # export data tests
+  describe '#fetch_export_data' do
+    include_context 'account'
+
+    context 'when fetching account from api' do
+
+      it 'when account id not found' do
+        expect{ account_class.send(:fetch_export_data, account_id_invalid) }.to raise_error(Interrupt, 'Account id not found')
+      end
+
+      it 'when account id found' do
+        expect{ account_class.send(:fetch_export_data, account_id) }.not_to raise_error(Interrupt, 'Account id not found')
+        expect(account_class.send(:fetch_export_data, account_id)).to match(account_id)
```

Comment:
This won't work unless the account exists in Kill Bill, right? If so, could you create it using the killbill-client-ruby library?